File: 1475.final-prices-with-a-special-discount-in-a-shop.js
// @leet start
/**
 * @param {number[]} prices
 * @return {number[]}
 */
var finalPrices = function(prices) {
	let answer = [];
	for (let i = 0; i < prices.length; i++) {
		const price = prices[i];
		var j = i + 1;
		while (j < prices.length && prices[j] > price) j++;
		if (j === prices.length) {
			answer.push(price);
		} else {
			answer.push(price - prices[j]);
		}
	}
	return answer;
};
// @leet end

File: 1475.final-prices-with-a-special-discount-in-a-shop.c
// @leet start
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int *finalPrices(int *prices, int pricesSize, int *returnSize) {
  int *answer = (int *)malloc(pricesSize * sizeof(int));
  *returnSize = pricesSize;
  for (int i = 0; i < pricesSize; i++) {
    int j = i + 1;
    while (j < pricesSize && prices[i] < prices[j])
      j++;
    if (j == pricesSize) {
      answer[i] = prices[i];
    } else {
      answer[i] = prices[i] - prices[j];
    }
  }
  return answer;
}
// @leet end
